File: src/services/schema-export/src/tests/demo-model.swift
import Foundation
import RealmSwift

class RealmTestClass0: Object {
    @objc dynamic var integerValue: Int = 0
    @objc dynamic var stringValue: String? = nil
    @objc dynamic var dataValue: Data? = nil
}

class RealmTestClass1: Object {
    @objc dynamic var integerValue: Int = 0
    @objc dynamic var boolValue: Bool = false
    @objc dynamic var floatValue: Float = 0
    @objc dynamic var doubleValue: Double = 0
    @objc dynamic var stringValue: String? = nil
    @objc dynamic var dateValue: Date? = nil
    let arrayReference = List<RealmTestClass0>()
}

class RealmTestClass2: Object {
    @objc dynamic var integerValue: Int = 0
    @objc dynamic var boolValue: Bool = false
    @objc dynamic var objectReference: RealmTestClass1?
}

File: src/services/schema-export/src/tests/demo-model.js
exports.RealmTestClass0 = {
  name: 'RealmTestClass0',
  properties: {
    integerValue: 'int',
    stringValue: 'string?',
    dataValue: 'data?'
  }
}

exports.RealmTestClass1 = {
  name: 'RealmTestClass1',
  properties: {
    integerValue: 'int',
    boolValue: 'bool',
    floatValue: 'float',
    doubleValue: 'double',
    stringValue: 'string?',
    dateValue: 'date?',
    arrayReference: 'RealmTestClass0[]'
  }
}

exports.RealmTestClass2 = {
  name: 'RealmTestClass2',
  properties: {
    integerValue: 'int',
    boolValue: 'bool',
    objectReference: 'RealmTestClass1'
  }
}

